coala_quickstart/generation/Project.py

```diff
@@ -29,3 +30,79 @@ def valid_path(path: StringConverter):
     if not os.path.isdir(path):
         raise ValueError("The given path doesn't exist.")
     return path
+
+
+def language_composition(file_paths):
+    """
+    Computes the composition of each language, with unknown extensions
+    tagged with the ``Unknown`` key.
+
+    :param file_paths: A list of file paths.
+    :return:           A dict with file name as key and the percentage
+                       of occurences as the value.
+    """
+    results = {lang: 0 for ext in exts for lang in exts[ext]}
+    results["Unknown"] = 0
+
+    for file_path in file_paths:
+        ext = os.path.splitext(file_path)[1]
+        if ext in exts:
+            for lang in exts[ext]:
+                results[lang] += 100 / len(file_paths)
+        else:
+            results["Unknown"] += 100 / len(file_paths)
+
+    return results
+
+
+def get_used_languages(file_paths):
+    """
+    Identifies the most used languages in the user's project directory
+    from the files matched from the given glob expression.
+
+    :param file_paths:
+        A list of absolute file paths in the user's project directory.
+    :return:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    results = language_composition(file_paths)
+
+    # Sort the languages based on frequency
+    languages = sorted(
+        results.items(),
+        key=operator.itemgetter(1),
+        reverse=True)
+
+    append_list = []
+    unknown_percent = 0
+
+    for lang, percent in languages:  # pragma: no branch
+        if percent > 0:
+            if lang == "Unknown":
+                unknown_percent = percent
```

Comment:
Because we want `Unknown` to be at the bottom. In general, `Unknown` takes about 25-30% in a project. So it doesn't look good for us to put it in the second place. So the first n-1 are sorted based on the percentage use and the last one, irrespective of the composition, is Unknown with (100 - sigma(rest))